numpy/core/fromnumeric.py

```diff
@@ -62,6 +61,32 @@ def _wrapfunc(obj, method, *args, **kwds):
         return _wrapit(obj, method, *args, **kwds)
 
 
+def _wrapreduction(obj, method, name, axis, dtype, out, **kwargs):
+    passkwargs = {}
+    for k, v in kwargs.items():
+        if v is not np._NoValue:
+            passkwargs[k] = v
+
+    if isinstance(obj, _gentype):
+        res = reduce(method, obj, method.identity)
+        if out is not None:
+            out[...] = res
+            return out
+        return res
+    if type(obj) is not mu.ndarray:
+        try:
+            reduction = getattr(obj, name)
```

Comment:
I actually think we want to deprecate this too - do we need `__getattr__`-style duck-typing now that we have `__array_ufunc__`?

Reply:
If a library wants to support both old and new versions of Numpy, it's needed.